process/block/shardinfo_test.go

```diff
@@ -0,0 +1,1125 @@
+		shardData, err := sic.createShardDataFromExecutionResult(execResult)
+		require.Equal(t, fmt.Errorf("GetHeaderByHash error"), err)
+		require.Nil(t, shardData)
+	})
+
+	t.Run("should fail when updateShardDataWithCrossShardInfo fails", func(t *testing.T) {
+		t.Parallel()
+		args := createDefaultShardInfoCreateDataArgs()
+		args.headersPool.GetHeaderByHashCalled = func(hash []byte) (data.HeaderHandler, error) {
+			return getShardHeaderForShard(uint32(1)), nil
+		}
+
+		sic, err := NewShardInfoCreateData(
+			args.enableEpochsHandler,
+			args.headersPool,
+			args.proofsPool,
+			args.pendingMiniBlocksHandler,
+			args.blockTracker,
+		)
+		require.Nil(t, err)
+		execResult := getExecutionResultForShard(uint32(1), []byte("header hash for shard 1"))
+		shardData, err := sic.createShardDataFromExecutionResult(execResult)
+		require.Equal(t, fmt.Errorf("notarized headers slice is nil"), err)
+		require.Nil(t, shardData)
+	})
+
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+		expectedNonce := uint64(1)
+		header := getShardHeaderForShard(uint32(1))
+		args := createDefaultShardInfoCreateDataArgs()
+		args.headersPool.GetHeaderByHashCalled = func(hash []byte) (data.HeaderHandler, error) {
+			return header, nil
+		}
+		args.pendingMiniBlocksHandler.GetPendingMiniBlocksCalled = func(shardID uint32) [][]byte {
+			return [][]byte{[]byte("hash1"), []byte("hash2")}
+		}
+		args.blockTracker.GetLastSelfNotarizedHeaderCalled = func(shardID uint32) (data.HeaderHandler, []byte, error) {
+			return &block.Header{Nonce: expectedNonce}, []byte("selfNotarizedHash"), nil
+		}
+		sic, err := NewShardInfoCreateData(
+			args.enableEpochsHandler,
+			args.headersPool,
+			args.proofsPool,
+			args.pendingMiniBlocksHandler,
+			args.blockTracker,
+		)
+		require.Nil(t, err)
+		execResult := getExecutionResultForShard(uint32(1), []byte("header hash for shard 1"))
+		shardData, err := sic.createShardDataFromExecutionResult(execResult)
+		require.Nil(t, err)
+		require.NotNil(t, shardData)
+		assert.Equal(t, uint32(2), shardData.GetNumPendingMiniBlocks())
+		assert.Equal(t, uint32(execResult.GetExecutedTxCount()), shardData.GetTxCount())
+		assert.Equal(t, uint32(1), shardData.GetShardID())
+		assert.Equal(t, execResult.GetAccumulatedFees(), shardData.GetAccumulatedFees())
+		assert.Equal(t, execResult.GetHeaderHash(), shardData.GetHeaderHash())
+		assert.Equal(t, execResult.GetHeaderRound(), shardData.GetRound())
+		assert.Equal(t, header.GetPrevHash(), shardData.GetPrevHash())
+		assert.Equal(t, execResult.GetHeaderNonce(), shardData.GetNonce())
+		assert.Equal(t, header.GetPrevRandSeed(), shardData.GetPrevRandSeed())
+		assert.Equal(t, header.GetPubKeysBitmap(), shardData.GetPubKeysBitmap())
+		assert.Equal(t, execResult.GetAccumulatedFees(), shardData.GetAccumulatedFees())
+		assert.Equal(t, execResult.GetDeveloperFees(), shardData.GetDeveloperFees())
+		require.Equal(t, header.GetEpoch(), shardData.(*block.ShardData).GetEpoch())
```

Comment:
[optional]

As a general remark, it would perhaps be cleaner if we would only use one testing library instead of a mix between assert & require